### PR TITLE
V2

### PR DESCRIPTION
**Pull request by manelmontilla**
This PR major refactor in the vulcan-agent.
The refactor aims to simplify the implementation and improve the reliability of the agent by leveraging the agreement of the team in a new set of requirements for the agent, and the whole vulcan-core. Concretely:  

- We should be able to continuously run a big number of checks in a reliable way. 

- The Vulcan Core will provide external systems a way to know the status of a check, both using poll and push (http endpoint and SNS).

- Each check generates a report that can contain one or more vulnerabilities. Each vulnerability can contain large chunks of data related to descriptions and other resources. 

- We also want to keep the logs, that is the stdout and stderr, generated by a check running in its container. Those logs are also potentially big. Those numbers make it impossible, or not a good practice, to store those reports and logs directly in sqs, or even in any other queue or service bus like Kafka.

- We should be able to abort a check at any time, that is: kill a check(container) that is running or prevent initiating a check that is aborted. (Not implemented yet)

- An agent will stop listening and reading from a queue when a configurable amount of time  has passed without getting any message from the queue it is listening on.
 
- A check may run more than once.

### CODE REVIEW

File: metrics/metrics.go [hunk 1]
```diff
@@ -0,0 +1,127 @@
...
+	if !p.Enabled {
+		return
+	}
+	metrics := metrics.Metric{
+		Name:  "vulcan.stream.mssgs.received",
+		Typ:   metrics.Count,
+		Value: 1,
+		Tags: []string{
+			"component:agent",
+			fmt.Sprint("action:", "abort"),
```
Comment (ka3de) [inline metrics/metrics.go:122]: Very silly, but no need for the `fmt.Sprint` as you know.

**Review by ka3de — COMMENTED** [review]
Awesome job @manelmontilla , I really liked it! :1st_place_medal: 

Just a couple of very silly comments. Also, the `_resources/example.toml` file is empty, I guess it should contain a config template? And also, I think the `README` needs update.

**Comment by manelmontilla** [comment]
> Awesome job @manelmontilla , I really liked it! 🥇
> 
> Just a couple of very silly comments. Also, the `_resources/example.toml` file is empty, I guess it should contain a config template? And also, I think the `README` needs update.

Thanks!! and absolutely agree I will fix both issues!!

File: metrics/metrics.go (hunk 1 above)
Comment (manelmontilla) [reply metrics/metrics.go:122]: True, thanks!!

**Review by ka3de — APPROVED** [review]
LGTM.

**Review by ka3de — APPROVED** [review]
LGTM.

**Pull request merged**